pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -417,12 +417,27 @@ func (m *kubeGenericRuntimeManager) podSandboxChanged(pod *v1.Pod, podStatus *ku
 		}
 	}
 
-	// Needs to create a new sandbox when readySandboxCount > 1 or the ready sandbox is not the latest one.
+	// Needs to create a new sandbox when readySandboxCount > 1.
 	sandboxStatus := podStatus.SandboxStatuses[0]
 	if readySandboxCount > 1 {
 		klog.V(2).Infof("More than 1 sandboxes for pod %q are ready. Need to reconcile them", format.Pod(pod))
 		return true, sandboxStatus.Metadata.Attempt + 1, sandboxStatus.Id
 	}
+
+	//check the pod is succeeded
+	succeededContainerCount := 0
+	for _, container := range pod.Spec.Containers {
+		if containerSucceeded(&container, podStatus) {
```

Comment:
The logic for this method looks wrong to me... I think the container should only be considered succeeded if it's state is `ContainerStateExited`.

/cc @yujuhong

Reply:
Do you mean that there is a problem with the logic of the function of `containerSucceeded`?

Reply:
Yes. In the context we used it before, it didn't matter, but it would in this context.